### PR TITLE
feat: (RI-4198) Add onepassword provider

### PR DESCRIPTION
**Pull request by javiercm1410**

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: 📝 Lint
+
+on:
+  pull_request:
```
Comment (b-nicasio) [inline .github/workflows/lint.yml:4]: On pull request only to main, remember this is part of the TBD rules

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        run: npm install --only=dev
+
+        # Check all commits pushed to this PR
+      - name: 👁️‍🗨️ Validate PR commits with commitlint
+        run: >- 
+          npx commitlint
+          --from ${{ github.event.pull_request.head.sha }}~${{ github.event.pull_request.commits }}
+          --to ${{ github.event.pull_request.head.sha }}
+          --verbose
+      - name: 🔎 Validate PR title with commitlint
```
Comment (b-nicasio) [inline .github/workflows/lint.yml:53]: Missed an space here between steps

File: .husky/pre-commit [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+BRANCH=$(git rev-parse --abbrev-ref HEAD)
+REGEX="^(build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test)-[A-Z]{2,4}-[0-9]{1,5}-[A-Za-z0-9-]+$"
+
+if ! [[ $BRANCH =~ $REGEX ]]; then
+  echo "Your commit was rejected due to branching name"
+  exit 1
+fi
```
Comment (b-nicasio) [inline .husky/pre-commit:9]: Why don't we add the code lint here as well?

File: .github/workflows/lint.yml (hunk 1 above)
Comment (javiercm1410) [reply .github/workflows/lint.yml:4]: done

File: .husky/pre-commit (hunk 3 above)
Comment (javiercm1410) [reply .husky/pre-commit:9]: nice

**Review by angelmadames — CHANGES_REQUESTED** [review]
I'm confused as to why we removed the upload to S3? 👀 
Even if we are using releases , we can have an alternative backup/download mechanism.

File: src/parser.ts [hunk 4]
```diff
@@ -22,6 +22,6 @@ export function generateEnvFile(envObject: EnvVarObject, filePath: string) {
     .map(([key, value]) => `${key}='${value}'`)
     .join('\n');
 
-  fs.writeFileSync(filePath, envContent, 'utf8');
+  fs.appendFileSync(filePath, envContent + '\n', 'utf8');
```
[the remark this answers is not in the document]
Comment (javiercm1410) [reply src/parser.ts:25]: the new value will override the previous one on code, but both will be on the file

[the remark this answers is not in the document]
Comment (javiercm1410) [reply src/parser.ts]: This logic was added!

File: src/commands/op.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        integrationVersion: version,
+      });
+        
+      for (const item of items) {
+        const vaultItem = await client.items.get(vault, item);
+
+        const secret = vaultItem.fields.map(obj=> ({ title: obj.title, value: obj.value }));
+
+        const envData = secret.reduce(
+          (obj, item) => Object.assign(obj, { [item.title]: item.value }), {});
```
[the remark this answers is not in the document]
Comment (javiercm1410) [reply src/commands/op.ts]: Nice, done!

**Review by angelmadames — APPROVED** [review]

**Pull request merged**